api/config/config_perf.json

```diff
@@ -21,7 +21,8 @@
     "retryCount": 4,
     "bufferCapacity": 300,
     "bufferRefillProjectsPerTask": 5,
-    "defaultFreeCreditsLimit": 100.0,
```

Comment:
Ooh – interesting to see a PR that removes an existing WorkbenchConfig field. I'm not aware of any clear operational guidelines on how to do this safely (it's arguably somewhat similar to removing a database field, w.r.t. rollout and rollback safety). @calbach I'm curious to hear if there is some existing rule we've been following here – in which case I'd love to find a good place to document it.

Taking a quick dip into the details: when we deploy the Workbench, our scripts push the config (https://github.com/all-of-us/workbench/blob/1ab8006207ebc5e01dfa3059b18bfd63863b4167/api/libproject/devstart.rb#L1692) before deploying the App Engine app (https://github.com/all-of-us/workbench/blob/1ab8006207ebc5e01dfa3059b18bfd63863b4167/api/libproject/devstart.rb#L1709). Depending on how long the config is cached in-memory by tasks (and/or whether any new tasks are kicked off in-between these steps), and how the config-parsing machinery works, I think we may risk some transient app failures during this time window.

But I can't speak with confidence one way or another. I haven't heard of this happening in the past ~year, but also our user base on prod has been very low up until November.

Reply:
Hmm, yep I don't think we have any guidance on this yet. The usual pattern for flag removals is:

1. Send PR to eliminate all code references to the flag and document it as deprecated / pending removal
2. Wait for a release to go out (and wait beyond a reasonable doubt of needing to roll that back)
3. Send PR to remove the flag.

(1) is also a bit more nuanced than usual because our config update and code release do not happen atomically. So I think additionally, (1) should leave the prior config value in place s.t. it will still work in combination with the previous binary.

Reply:
👍 will restore the old value and make a comment